### PR TITLE
Feat/auth form validation

### PR DESCRIPTION
**Pull request by KorneyKorneich**
### Description

In this pull request, I've integrated `react-hook-form` into the project to manage form state and validation more efficiently. The authentication forms (`SignIn` and `SignUp`) have been refactored to utilize `react-hook-form` for better performance and simpler code structure.

### Changes

1. **Integration of `react-hook-form`:**
   - Added `react-hook-form` to the project dependencies.
   - Replaced existing form handling logic with `react-hook-form`'s `useForm` and `Controller` hooks.

2. **Refactored Authentication Forms:**
   - Refactored `SignIn` and `SignUp` components to use `react-hook-form` for managing input states, validations, and submissions.
   - Removed redundant state management and handler functions previously used for form control.

3. **Form Validation:**
   - Implemented client-side validation using `react-hook-form`'s validation rules.
   - Validation rules include:
     - Email format validation with a regex pattern.
     - Required field checks for email and password.
     - Password minimum length validation (minimum 6 characters).
   - Displayed validation error messages directly within the form UI.

4. **Error Handling:**
   - Enhanced error handling by mapping Firebase authentication errors to the respective form fields.
   - Displayed error messages for invalid email, weak passwords, and other relevant authentication errors.

### Benefits

- **Improved Performance:** `react-hook-form` minimizes re-renders and improves overall form performance.
- **Simplified Codebase:** Form management and validation are now more streamlined and maintainable.
- **Better UX:** Real-time validation and clear error messages improve user experience.

### CODE REVIEW

File: src/entities/user/model/userStore.ts [hunk 1]
```diff
@@ -2,67 +2,87 @@ import { create } from "zustand";
...
 	signUp: async (credentials: Credentials) => {
 		const { email, password } = credentials;
-		set({isLoading: true});
-		await auth().createUserWithEmailAndPassword(email, password)
+		set({ isLoading: true, loginError: null });
+		await auth()
+			.createUserWithEmailAndPassword(email, password)
 			.then((res) => {
 				set({
-					user:{
```
Comment (VladisAga) [inline src/entities/user/model/userStore.ts]: you have the same problems with tabs like me

File: src/screens/OTPVerification/ui/OTPVerification.tsx [hunk 2]
```diff
@@ -1,33 +1,30 @@
...
-			: navigation.goBack();
+		confirm ? OTPConfirm({ confirm, code }) : navigation.goBack();
 	};
 
 	return (
 		<SafeAreaView>
 			<KeyboardAvoidingView behavior={"height"} style={styles.container}>
 				<View style={styles.img}>
-					<VerificationImg height={200} width={200}/>
+					<VerificationImg height={200} width={200} />
```
Comment (VladisAga) [inline src/screens/OTPVerification/ui/OTPVerification.tsx]: styles should be in the style file

File: src/screens/PhoneAuthScreen/ui/PhoneAuthScreen.tsx [hunk 3]
```diff
@@ -35,17 +37,15 @@ export const PhoneAuthScreen = ({navigation}: UnauthorizedStackRoutesProps) => {
...
+				await phoneAuth({ selectedCountry: selectedCountry, phone: data.phone }).then((res) => {
+					navigation.navigate("OTPVerifyScreen", { confirmation: res });
+				});
 			}
-			navigation.navigate( "OTPVerifyScreen", { confirmation: confirmation } );
-		}catch (e){
-			//todo: add error handler
-		}
-	}
+		} catch (e) {}
```
Comment (VladisAga) [inline src/screens/PhoneAuthScreen/ui/PhoneAuthScreen.tsx]: add logic in the catch or remove it

File: src/widgets/SignUp/ui/SignUp.tsx [hunk 4]
```diff
@@ -1,23 +1,96 @@
...
-	const handleSignUp = () => {
-		signUp({email, password});
+			if (firebaseError.code === "auth/email-already-in-use" || firebaseError.code === "auth/invalid-email") {
+				setError("email", { type: "manual", message: errorMessage });
+			} else if (firebaseError.code === "auth/weak-password") {
+				setError("password", { type: "manual", message: errorMessage });
+			} else {
+				setError("email", { type: "manual", message: errorMessage });
+				setError("password", { type: "manual", message: errorMessage });
+			}
```
Comment (VladisAga) [inline src/widgets/SignUp/ui/SignUp.tsx]: use switch case

**Review by VladisAga — CHANGES_REQUESTED** [review]
Small changes

**Review by VladisAga — APPROVED** [review]

File: src/App/navigation/routes/unauthorizedStackRoutes.ts [hunk 5]
```diff
@@ -1,48 +1,35 @@
...
 import { OTPVerification } from "@screens/OTPVerification";
 import { StackNavigationOptions } from "@react-navigation/stack";
-import { AuthorizedStackParamList, UnauthorizedStackParamList, ROUTES } from "@shared/api";
-import { NativeStackScreenProps } from "react-native-screens/native-stack";
+import { UnauthorizedStackParamList, ROUTES } from "@shared/api";
 
 type RouteConfig<ParamList extends Record<string, object | undefined>, RouteName extends keyof ParamList> = {
     name: RouteName;
-    component: React.ComponentType<NativeStackScreenProps<ParamList, RouteName>>;
+    component: React.ComponentType<any>; // Expect a component, not an element
```
Comment (VladisAga) [inline src/App/navigation/routes/unauthorizedStackRoutes.ts]: what about this any, can you change this on some certain type?

File: src/screens/PhoneAuthScreen/ui/PhoneAuthScreen.tsx [hunk 6]
```diff
@@ -1,50 +1,54 @@
...
+    function handleModalClose() {
         setIsModalVisible(false);
-    };
+    }
 
-    const handleModalOpen = () => {
+    function handleModalOpen() {
         setIsModalVisible(true);
-    };
+    }
```
Comment (VladisAga) [inline src/screens/PhoneAuthScreen/ui/PhoneAuthScreen.tsx]: why did you remove arrow func?

File: src/screens/PhoneAuthScreen/ui/PhoneAuthScreen.tsx [hunk 7]
```diff
@@ -1,50 +1,54 @@
...
-            confirmation: confirmation,
-        });
     };
 
-    const handleCountryPick = (item: Country) => {
+    function handleCountryPick(item: Country) {
         setSelectedCountry(item);
         setIsModalVisible(false);
-    };
+    }
```
Comment (VladisAga) [inline src/screens/PhoneAuthScreen/ui/PhoneAuthScreen.tsx:50]: why did you remove arrow func?

**Review by VladisAga — APPROVED** [review]

File: src/App/navigation/routes/unauthorizedStackRoutes.ts (hunk 5 above)
Comment (KorneyKorneich) [reply src/App/navigation/routes/unauthorizedStackRoutes.ts]: nothing but <any> fits here
